homeassistant/components/panasonic_viera/__init__.py

```diff
@@ -203,47 +199,51 @@ async def async_send_key(self, key: Keys | str) -> None:
         except (AttributeError, TypeError):
             key = getattr(key, "value", key)
 
+        assert self._control is not None
         await self._handle_errors(self._control.send_key, key)
 
-    async def async_turn_on(self, context):
+    async def async_turn_on(self, context: Context | None) -> None:
         """Turn on the TV."""
         if self._on_action is not None:
             await self._on_action.async_run(context=context)
             await self.async_update()
-        elif self.state != STATE_ON:
+        elif self.state != MediaPlayerState.ON:
             await self.async_send_key(Keys.POWER)
             await self.async_update()
 
-    async def async_turn_off(self):
+    async def async_turn_off(self) -> None:
         """Turn off the TV."""
-        if self.state != STATE_OFF:
+        if self.state is not MediaPlayerState.OFF:
             await self.async_send_key(Keys.POWER)
-            self.state = STATE_OFF
+            self.state = MediaPlayerState.OFF
             await self.async_update()
 
-    async def async_set_mute(self, enable):
+    async def async_set_mute(self, enable: bool) -> None:
         """Set mute based on 'enable'."""
+        assert self._control is not None
         await self._handle_errors(self._control.set_mute, enable)
 
-    async def async_set_volume(self, volume):
+    async def async_set_volume(self, volume: int) -> None:
         """Set volume level, range 0..1."""
+        assert self._control is not None
         volume = int(volume * 100)
         await self._handle_errors(self._control.set_volume, volume)
 
-    async def async_play_media(self, media_type, media_id):
+    async def async_play_media(self, media_type: MediaType, media_id: str) -> None:
         """Play media."""
+        assert self._control is not None
         _LOGGER.debug("Play media: %s (%s)", media_id, media_type)
         await self._handle_errors(self._control.open_webpage, media_id)
 
-    async def async_get_device_info(self):
+    async def async_get_device_info(self) -> dict[str, Any] | None:
         """Return device info."""
         if self._control is None:
             return None
-        device_info = await self._handle_errors(self._control.get_device_info)
+        device_info = await self._handle_errors(self._control.get_device_info)  # type: ignore[func-returns-value]
```

Comment:
```suggestion
        device_info = await self._handle_errors(self._control.get_device_info)
```
Not necessary anymore.